public/react/Configure/components/Teams/components/Select/Select.tsx

```diff
@@ -27,7 +33,13 @@ const Select: FC<SelectProps> = ({ name, control, label, jiraFields, onSave, dis
       control={control}
       disabled={disabled}
       render={({ field }) => {
-        const selectedOption = jiraFields.find((option) => {
+        const allFields = isSingle(jiraFields)
+          ? jiraFields
+          : jiraFields.reduce((all, group) => {
```

Comment:
You're not using the "label" field for groups when you flatten here, so instead of
```
jiraFields: Array<{ label: string; value: string }>
    | Array<{ label: string; options: Array<{ label: string; value: string }> }>;
```
 couldn't this have just been 

```
type SelectField = { label: string; value: string } | SelectField[] ;
...
jiraFields: SelectField[];
```

Reply:
Then at this line,
```
jiraFields.flat()
```

Reply:
I'm not really following the train of thought (Sorry been a long day), we don't control the shape so we can simplify the typing to something like:

```ts
  jiraFields: Array<SelectField> | Array<{ label: string; options: Array<SelectField> }>;
```

but the shape needs to stay. Now can we use flat? Probably but I'm not following how you're intending it to work

Reply:
oh also, The label is used by the underlying component

Reply:
Do you mean something like:
```ts
const allFields = isSingle(jiraFields)
  ? jiraFields
  : jiraFields.flatMap(group => group.options);
```

Reply:
OK, that makes sense, Line 52 where the unflattened options are being passed in is what I missed.  Approving.